modules/ensemble/lib/widget/input/form_textfield.dart

```diff
@@ -346,11 +346,22 @@ class TextInputState extends FormFieldWidgetState<BaseTextInput>
   void didUpdateWidget(covariant BaseTextInput oldWidget) {
     super.didUpdateWidget(oldWidget);
     widget.controller.inputFieldAction = this;
+
+    // Making sure to move cursor to end when widget rebuild
+    // issue: https://github.com/EnsembleUI/ensemble/issues/1527
+    int cursorPosition = widget.textController.selection.baseOffset;
+    if (widget.textController.text.length != cursorPosition &&
+        focusNode.hasFocus) {
+      widget.textController.selection = TextSelection.fromPosition(
+        TextPosition(offset: widget.textController.text.length),
```

Comment:
I don't understand - wouldn't this move the cursor to the very end even when the user has it placed in the middle of the text while editing it? Isn't that the actual bug?

Reply:
No, the bug is when textfield rebuild which happens when textfield in bottomModal, the cursor starts from the end. 
The reason behind it, essentially flutter correctly puts at the end but bottomModal append bottomInsets when keyboard is displayed which makes our ensemble widget rebuild and put cursor again in starting position. 

The solution just make sure to put it at the end if required.

Reply:
A better behavior would be to get the previous cursor position and go to that exact position. If the old text is longer somehow (which it shouldn't) the go to the end. You already have access to the old controller